cmd/crank/xpkg/install.go

```diff
@@ -50,27 +50,44 @@ const (
 	errKubeClient    = "failed to create kube client"
 )
 
-// InstallCmd is exported so that it can be re-used by the beta xpkg subcommand.
+// installCmd installs a package.
+type installCmd struct {
+	// Arguments.
+	Kind  string `arg:"" help:"The kind of package to install. One of \"provider\", \"configuration\", or \"function\"." enum:"provider,configuration,function"`
+	Image string `arg:"" help:"The OCI image name of the package to install."`
+	Name  string `arg:""  optional:"" help:"The name of the new package resource in the Crossplane API. Derived from the package's OCI image name by default."`
+
+	// Flags. Keep sorted alphabetically.
+	RuntimeConfigName    string        `placeholder:"NAME" help:"The name of the runtime config the package should use."`
+	ManualActivation     bool          `short:"m" help:"Require the new package's first revision to be manually activated."`
+	PackagePullSecrets   []string      `placeholder:"NAME" help:"A comma-separated list of secret names the package manager should use to pull the package from the registry."`
+	RevisionHistoryLimit int64         `short:"r" default:"LIMIT" help:"How many package revisions may exist before the oldest revisions are deleted."`
+	Wait                 time.Duration `short:"w" default:"0s" help:"How long to wait for the package to install before returning. The command does not wait by default."`
+}
```

Comment:
This renders as:

```
$ go run cmd/crank/main.go xpkg install --help
Usage: crossplane xpkg install <kind> <image> [<name>]

Install a package from a package registry.

Crossplane can be extended using packages. A Crossplane package is sometimes
called an xpkg. Crossplane supports configuration, provider and function
packages.

A package is an opinionated OCI image that contains everything needed to extend
Crossplane with new functionality. For example installing a provider package
extends Crossplane with support for new kinds of managed resource (MR).

This command tells the Crossplane package manager to install a package from a
package registry. It uses ~/.kube/config to know how to connect to the package
manager. You can override this using the KUBECONFIG environment variable.

See https://docs.crossplane.io/latest/concepts/packages for more information.

Arguments:
  <kind>      The kind of package to install. One of "provider",
              "configuration", or "function".
  <image>     The OCI image name of the package to install.
  [<name>]    The name of the new package resource in the Crossplane API.
              Derived from the package's OCI image name by default.

Flags:
  -h, --help                 Show context-sensitive help.
      --verbose              Print verbose logging statements.
  -v, --version              Print version and quit.

      --runtime-config-name=NAME
                             The name of the runtime config the package should
                             use.
  -m, --manual-activation    Require the new package's first revision to be
                             manually activated.
      --package-pull-secrets=NAME,...
                             A comma-separated list of secret names the package
                             manager should use to pull the package from the
                             registry.
  -r, --revision-history-limit=LIMIT
                             How many package revisions may exist before the
                             oldest revisions are deleted.
  -w, --wait=0s              How long to wait for the package to install before
                             returning. The command does not wait by default.
```

Note that I've opted not to explicitly state that only functions and providers support runtime config here. If someone tries to specify a runtime config when installing a configuration they'll get an error telling them its not supported.

I've also opted to say runtime config, not DeploymentRuntimeConfig, because we don't think DeploymentRuntimeConfig will always be the only type of runtime config. The API supports adding other types in future.